featuretools/feature_discovery/LiteFeature.py

```diff
@@ -40,37 +41,134 @@ def __init__(
         related_features: Optional[Set[LiteFeature]] = None,
         idx: Optional[int] = None,
     ):
-        self.name = name
-        self.logical_type = logical_type
-        self.tags = tags if tags else set()
-        self.primitive = primitive
-        self.base_features = base_features if base_features else []
-        self.df_id = df_id
-        self.idx = idx if idx is not None else 0
-        self.related_features = related_features if related_features else set()
-
-        if self.primitive:
-            assert isinstance(self.primitive, PrimitiveBase)
+        self._name = name
+        self._logical_type = logical_type
+        self._tags = tags if tags else set()
+        self._primitive = primitive
+        self._base_features = base_features if base_features else []
+        self._df_id = df_id
+        self._idx = idx if idx is not None else 0
+        self._related_features = related_features if related_features else set()
+
+        if self._primitive:
+            assert isinstance(self._primitive, PrimitiveBase)
             assert (
                 len(self.base_features) > 0
             ), "there must be base features if given a primitive"
-            self.n_output_features = self.primitive.number_output_features
-            self.depth = max([x.depth for x in self.base_features]) + 1
-            self._gen_name = self.primitive.generate_name(
-                [x.get_name() for x in self.base_features],
+            self._n_output_features = self._primitive.number_output_features
+            self._depth = max([x.depth for x in self.base_features]) + 1
+            self._alias = self._name
+            self._name = self._primitive.generate_name(
+                [x.name for x in self.base_features],
             )
 
-        elif self.name is None:
+        elif self._name is None:
             raise TypeError("Name must be given if origin feature")
-        elif self.logical_type is None:
+        elif self._logical_type is None:
             raise TypeError("Logical Type must be given if origin feature")
-        else:
-            self._gen_name = self.name
 
-        if self.logical_type is not None and "index" not in self.tags:
-            self.tags = self.tags | self.logical_type.standard_tags
+        if self._logical_type is not None and "index" not in self._tags:
+            self._tags = self._tags | self._logical_type.standard_tags
+
+        self._id = self._generate_hash()
+
+    @property
+    def name(self):
+        if self._alias:
+            return self._alias
+        elif self.is_multioutput():
+            return f"{self._name}[{self.idx}]"
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._alias = value
```

Comment:
I think it's confusing to be able to set `self._name` with the `name` param at init but set `self._alias` at this setter and have the property be able to return the alias or the name. Why not keep them totally separate fields?

Reply:
@thehomebrewnerd @tamargrey. I changed this allow users to set the alias with `set_alias`. This allows you to set the alias, but the getter on `name` still works the same. If an alias is present, it uses that, otherwise uses the name. Hope that addresses your concerns